optimum/amd/ryzenai/modeling.py

```diff
@@ -136,6 +128,9 @@ def __init__(
             **kwargs,
         )
 
+        if config:
+            self.model_type = config.model_type
```

Comment:
`config` is hinted as `PretrainedConfig`, not `Optional[PretrainedConfig]` so I am not sure about the controflow here.

In general for simplicity, I think we should probably avoid dynamically defined instance attributes (i.e., all `RyzenAIModel` instances should (or should not) have a `model_type` attribute.

Reply:
I guess I added it because some of their pretrained models do not have config. So I made two changes now:
1. Made config a hard requirement for `RyzenAIModelForCausalLM`
2. Made it optional in `RyzenAIModel`